File: apps/api/src/main/java/com/lifttrack/api/domain/models/RoutineExercise.java
package com.lifttrack.api.domain.models;

import java.time.Instant;
import java.util.UUID;

public record RoutineExercise(
        UUID uuid,
        UUID routineDayUuid,
        UUID exerciseUuid,
        Integer targetSets,
        String targetRepRange,
        int sortOrder,
        String notes,
        Instant createdAt) {

    public RoutineExercise create(UUID routineDayUuid, UUID exerciseUuid, Integer targetSets, String targetRepRange,
            int sortOrder, String notes) {
        if (routineDayUuid == null) {
            throw new IllegalArgumentException("Routine day UUID cannot be null");
        }
        if (exerciseUuid == null) {
            throw new IllegalArgumentException("Exercise UUID cannot be null");
        }
        if (targetSets != null && targetSets < 0) {
            throw new IllegalArgumentException("Target sets cannot be negative");
        }
        if (targetRepRange != null && targetRepRange.isBlank()) {
            throw new IllegalArgumentException("Target rep range cannot be blank");
        }
        if (sortOrder < 0) {
            throw new IllegalArgumentException("Sort order cannot be negative");
        }

        return new RoutineExercise(null, routineDayUuid, exerciseUuid, targetSets, targetRepRange, sortOrder, notes, null);
    }
}

File: apps/web/src/domain/models/routine-exercise.ts
export interface RoutineExercise {
  readonly uuid: string
  readonly routineDayUuid: string
  readonly exerciseUuid: string
  readonly targetSets: number | null
  readonly targetRepRange: string | null
  readonly sortOrder: number
  readonly notes: string | null
  readonly createdAt: string
}

export interface CreateRoutineExercise {
  readonly exerciseUuid: string
  readonly targetSets?: number
  readonly targetRepRange?: string
  readonly sortOrder: number
  readonly notes?: string
}
